### ISSUE TITLE
Automate HTTPS certificate renewal with Let's Encrypt

### ISSUE THREAD

**Issue by jankeromnes** [jankeromnes/janitor#33]
The `Makefile` currently offers a [way](https://github.com/jankeromnes/janitor/blob/master/Makefile#L86) to generate self-signed HTTPS certificates with `make https`, which is [used](https://github.com/jankeromnes/janitor/blob/master/Makefile#L9) in `make install`. (It also offers a [way](https://github.com/jankeromnes/janitor/blob/master/Makefile#L96) to just generate a CSR with `make https.csr`, in order to manually get it signed by a trusted CA.)

This should be deprecated in favour of automated Let's Encrypt renewal. The code for this is already [in](https://github.com/jankeromnes/janitor/blob/master/app.js#L27) [place](https://github.com/jankeromnes/janitor/blob/master/lib/certificates.js#L85), and just needs to be automated.

**Comment by jankeromnes** [comment]
An initialization task that can automatically renew HTTPS certificates was added in f9cde5199c0b178ddcba135853356fec5b5cc19d, and the cluster registration script `join.js` added in a55ee193bcfce2494e1fdd7436114ffb09cb83bf is using it. Now we just need to make the main server script `app.js` use it, and also switch the targeted LetsEncrypt service from staging to production.

**Comment by jankeromnes** [comment]
This is now fixed!

The primary server script `app.js` started running the automated HTTPS certificates boot task since fe680624d99b68ae3cdb354501c360177f2bd0d3, and 47e7b8d56cf0d23ee9e1fae5712a3fd6eeb99753 switched the targeted LetsEncrypt service from staging to production.

(Fun fact: this last commit was made from https://c9.io on my phone, in a very isolated chalet in the Swiss Alps, a few hours before New Year's Eve because the live certificate was about to expire, and obviously I broke the whole Janitor service by accidentally pushing work-in-progress commits from my phone into production, and then spent 2 hours in hell trying to fix everything again. Lesson learned: The "never deploy code on Fridays" rule also applies to New Year's Eve.)
